server/src/main/java/io/crate/replication/logical/MetadataTracker.java

```diff
@@ -136,23 +140,31 @@ private void run() {
                 clusterService.submitStateUpdateTask("track-metadata", new ClusterStateUpdateTask() {
                     @Override
                     public ClusterState execute(ClusterState localClusterState) throws Exception {
-                        return updateMetadata(subscriptionName, localClusterState, remoteClusterState);
+                        var result = updateMetadata(subscriptionName, localClusterState, remoteClusterState);
+                        countDown.onSuccess();
```

Comment:
This is not complete safe as we should wait until a possible new cluster state is published before triggering a new poll. Because `ClusterStateUpdateTask.clusterStatePublished()` is only called if the state changed, while we want to wait until the whole process is done despite of a new state, we must pass in a `ClusterStateTaskListener` to get a callback once all is done.

Reply:
Ok, I thought this is the case here. The reschedule is executed once `countDown.onSuccess()` is called n-times. E.g. if we have 3 subscriptions we need to wait until  `onSucess()` was called 3 times before `thenRun()` is executed. Anyway, I can have a look  `ClusterStateTaskListener`.

Reply:
`countDown.onSuccess()` is currently called after the cluster state is modified but *before*  the modified state is published locally and to all other nodes in the cluster.

Reply:
Ok, got it. Thanks for the feedback!